File: frontend/src/api/stats.ts
import { apiFetch } from "./http";

export type NamedCount = { id: string; name: string; count: number };
export type StatsSummary = {
  notes_total: number;
  starred_total: number;
  tags: NamedCount[];
  topics: NamedCount[];
};

export async function getStatsSummary(): Promise<StatsSummary> {
  return apiFetch<StatsSummary>("/api/stats/summary");
}

File: backend/app/schemas/stats.py
from pydantic import BaseModel


class NamedCount(BaseModel):
    id: str
    name: str
    count: int


class StatsSummaryResponse(BaseModel):
    notes_total: int
    starred_total: int
    tags: list[NamedCount]
    topics: list[NamedCount]

